app/controllers/discover.js

```diff
@@ -216,9 +216,6 @@ export default ApplicationController.extend({
     },
 
     search() {
-        if (!this.get('firstLoad')) {
```

Comment:
I think this is what resets the page when you change the query. If you're on page 5 and enter a new search, does it go to page 5 of the new results?